### PR TITLE
FIX: Admin sidebar was hiding chat/forum toggle button

### PR DESCRIPTION
**Pull request by martin-brennan**
We have separated and combined modes for sidebar panels.
Separated means the panels show only their own sections,
combined means sections from all panels are shown.

The admin sidebar only shows its own panels, so it must set
the mode to separated; however when we navigate to chat or
home we must revert to the initial mode setttings.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/routes/admin.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
     return I18n.t("admin_title");
   }
 
   activate() {
     if (this.currentUser.use_admin_sidebar) {
+      this.initialSidebarState = {
+        mode: this.sidebarState.mode,
+        displaySwitchPanelButtons: this.sidebarState.displaySwitchPanelButtons,
+      };
+
```
Comment (tgxworld) [inline app/assets/javascripts/admin/addon/routes/admin.js:28]: I actually wonder if the `sidebarState` service can track the "last" state and then API users can just call `revert` to previous state when they want to navigate away. 

cc @lis2

Comment (lis2) [reply app/assets/javascripts/admin/addon/routes/admin.js:28]: Interesting concept. So far we have 3 panels, "main", "chat" and "admin" but assume that in the future we will have more. We can remember the last mode per panel and as you suggested keep it in `sidebarState`.

Each time we change panel we can check if there is a history state first for the new panel.

Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/routes/admin.js:28]: Yes that could be another way of approaching it too :+1: We haven't really needed it in other situations until now.

Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/routes/admin.js:28]: Though, I would at least like to get this fix out then we can follow up with a nicer way of doing it.

**Review by tgxworld — APPROVED** [review]

File: app/assets/javascripts/admin/addon/routes/admin.js (hunk 1 above)
Comment (tgxworld) [reply app/assets/javascripts/admin/addon/routes/admin.js:28]: > we can follow up with a nicer way of doing it.

Absolutely 👍 Does @lis2 have the capacity to do this? 😁

**Pull request merged**